js/searchbar/customBangs.js

```diff
@@ -82,47 +200,84 @@ function initialize () {
     }
   })
 
-  // returns a task with the same name or index ("1" returns the first task, etc.)
-  function getTaskByNameOrNumber (text) {
-    const textAsNumber = parseInt(text)
+  bangsPlugin.registerCustomBang({
+    phrase: '!movetotask',
+    snippet: l('moveToTask'),
+    isAction: false,
+    showSuggestions: function (text, input, event) {
+      searchbarPlugins.reset('bangs')
+
+      var sortLastActivity = searchAndSortTasks(text)
 
-    return tasks.find((task, index) => (task.name && task.name.toLowerCase() === text) || index + 1 === textAsNumber
-    )
-  }
+      sortLastActivity.forEach(function (t) {
+
+        var task = t.task
+        var lastActivity = t.lastActivity
+
+        if (task.id != tasks.getSelected().id) {
+          var taskName = (task.name ? task.name : l('defaultTaskName').replace('%n', tasks.getIndex(task.id) + 1))
+
+          var data = {
+            title: taskName,
+            secondaryText: formatRelativeDate(lastActivity),
+            fakeFocus: false,
+            click: function () {
+              tabEditor.hide()
+              moveToTask('%n'.replace('%n', tasks.getIndex(task.id) + 1))
```

Comment:
This is confusing - we have a task ID already, we're converting it to a string, and then we're searching through all the tasks to convert that back into an ID. Also, it's possible to have multiple tasks with the same name, and this might cause us to pick a task that's different than what the user clicked on.

How about:
* Move the get/create task logic in `moveToTask` into its own function.
* Then moveToTask / switchToTask can take a task ID, and the callers can do `moveToTask(getTaskByNameAndNumber(text))` or `moveToTask(addOrCreateTask(text))` where necessary.

Reply:
Oh yes of course. It's a bit clunky. I had it in my head that by using the index (found by id and not name) would protect against duplicate names. But it is very clearly messy.